shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -207,28 +202,22 @@ impl AtomicBrokerId {
 
 #[derive(Debug)]
 struct SaslStatus {
```

Comment:
Should we just replace this with a simple `auth_complete: bool` field in KafkaSinkCluster?